### PR TITLE
feat: filter artworks to a sale

### PR DESCRIPTION
**Pull request by joeyAghion**
Like https://github.com/artsy/helix/pull/452, this adds the ability to filter to a given **sale**'s artworks for batch genoming purposes. Once again it mostly copies handling of fairs. Looks like:

Screenshot 2024-08-21 at 2 13 27 PM

### CODE REVIEW

File: app/javascript/components/batch_update/components/SearchForm.js [hunk 1]
```diff
@@ -111,6 +113,7 @@ class SearchForm extends React.Component {
           onRemoveTag={onRemoveTag}
           onRemoveArtist={onRemoveArtist}
           partner={partner}
+          sale={sale}
```
Comment (anandaroop) [inline app/javascript/components/batch_update/components/SearchForm.js:116]: **suggestion:** This is on the right track, but we need to thread this a little further [and replicate this](https://github.com/artsy/rosalind/blob/8dac405e64f7d1164eb2b1b2211ae4decf10ebb7/app/javascript/components/batch_update/components/CurrentCriteria.js#L65) in order to have the current sale criterion (and dismiss button) actually show up in the UI.

Comment (anandaroop) [reply app/javascript/components/batch_update/components/SearchForm.js:116]: **suggestion:** add test coverage as well, [like so](https://github.com/artsy/rosalind/blob/8dac405e64f7d1164eb2b1b2211ae4decf10ebb7/app/javascript/components/batch_update/components/CurrentCriteria.spec.js#L80-L85).

File: app/javascript/components/batch_update/components/SearchForm.js [hunk 2]
```diff
@@ -124,6 +127,7 @@ class SearchForm extends React.Component {
         />
         {partner === null && <PartnerAutosuggest updateState={updateState} />}
         {fair === null && <FairAutosuggest updateState={updateState} />}
+        {sale === null && <SaleAutosuggest updateState={updateState} />}
```
Comment (anandaroop) [inline app/javascript/components/batch_update/components/SearchForm.js:130]: **suggestion:** Add test coverage, [like so](https://github.com/artsy/rosalind/blob/8dac405e64f7d1164eb2b1b2211ae4decf10ebb7/app/javascript/components/batch_update/components/SearchForm.spec.js#L58-L63)

**Review by anandaroop — CHANGES_REQUESTED** [review]
I highlighted the key change below that will add the crucial UI behavior of giving you feedback about the selected sale and the ability to dismiss.

Directly analagous to how it works for fairs:

https://github.com/user-attachments/assets/34bf94e9-be46-46d2-9206-9a67a5ebc765

Also left a few suggestions for specs. Dilemma because the patterns are old, but I figure better to maintain coverage — and it should be fairly simple copypasta,

**Comment by joeyAghion** [comment]
Thank you 🙇 for the detailed advice. Those changes and tests were straightforward to add. (I actually sort of like the simplicity and repetition of Rosalind-era tooling.)

The new dismissable sale element looks like:
Screenshot 2024-08-23 at 11 13 48 AM

**Review by anandaroop — APPROVED** [review]
LGTM!

**Pull request merged**